packages/yew-macro/src/html_tree/html_component.rs

```diff
@@ -210,7 +228,30 @@ impl HtmlComponent {
                 },
             }),
             cursor,
-        ))
+        )
+    }
+
+    fn peek_component_type(cursor: Cursor) -> Option<(Type, Cursor)> {
```

Comment:
Now that we have the other changes, can we get rid of this method entirely?

Reply:
I'll have to refactor the entire `Parse` implementation for `HtmlComponent` then because it relies on this method. That's fine with me. Just that it will take some more time.

Reply:
You don't need to go totally overboard here. I wrote down [in an earlier comment](https://github.com/yewstack/yew/pull/1738#issuecomment-778312573) how we can solve this.

To be clear, my goal for this PR was to fix this exact issue, refactoring the peek logic of `HtmlTree` is a very much appreciated extra, but it doesn't solve the underlying issue here in my opinion.

If you want, we can also split this into two separate PRs to simplify things. What you've done here is already great as it is :)